### PR TITLE
fix: remove ServerAnimationHandler OnNetworkSpawn workaround [MTT-3152]

### PR DESCRIPTION
**Pull request by fernando-cortez**
### Description
This PR removes a previous workaround that previously synced life state for players. A joining player would not see a fully synced Animator. The cases where this was obvious was when a late joiner would not see fainted players in their fainted animation. This has since been fixed on the NetworkAnimator side and so this workaround is not necessary.

### Issue Number(s)
[MTT-3152](https://jira.unity3d.com/browse/MTT-3152)

### Contribution checklist
 - [ ] Tests have been added for boss room and/or utilities pack
 - [ ] Release notes have been added to the [project changelog](../CHANGELOG.md) file and/or [package changelog](../Packages/com.unity.multiplayer.samples.coop/CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] JIRA ticket ID is in the PR title or at least one commit message
 - [ ] Include the ticket ID number within the body message of the PR to create a hyperlink
 - [ ] An Index entry has been added in readme.md if applicable

### CODE REVIEW

**Comment by fernando-cortez** [comment]
Threw this PR up but it should wait until after [this PR](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop/pull/782) goes through for the 2.0.3 release.

File: Assets/Scripts/Gameplay/GameplayObjects/Character/ServerAnimationHandler.cs [hunk 1]
```diff
@@ -1,5 +1,4 @@
 using System;
```
Comment (SamuelBellomo) [inline Assets/Scripts/Gameplay/GameplayObjects/Character/ServerAnimationHandler.cs:1]: note release note?

**Comment by fernando-cortez** [comment]
We can't quite yet remove the workaround, since applying a change to the NetworkAnimator on OnNetworkSpawn() will currently throw NullReferenceExceptions.

This [PR](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/2309/files) addresses that issue, meaning this PR will go on hold for now.

**Comment by fernando-cortez** [comment]
[This PR](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop/pull/886) has included the work that this PR was aiming to initially accomplish, and so I'll close this PR.

**Pull request closed without merging**